### PR TITLE
D4Science profile builder

### PR DESCRIPTION
**Pull request by enolfc**
# Summary

Create profile list according to the information system in D4Science

---

**Related issue :**

### CODE REVIEW

File: egi_notebooks_hub/d4science.py [hunk 1]
```diff
@@ -37,6 +38,13 @@
 D4SCIENCE_OIDC_URL = os.environ.get(
     "D4SCIENCE_OIDC_URL", "https://accounts.d4science.org/auth/realms/d4science/"
 )
+D4SCIENCE_INFOSYS_URL = os.environ.get(
```
Comment (andrea-manzi) [inline egi_notebooks_hub/d4science.py:41]: We have already a D4SCIENCE_DM_REGISTRY_URL which is basically pointing to another path of the icproxy..maybe you can just make configurable the base path

**Comment by andrea-manzi** [comment]
i have seen new requests from D4science in the meantime...let's wait to see if some other changes are needed before completing the review....

**Comment by enolfc** [comment]
> i have seen new requests from D4science in the meantime...let's wait to see if some other changes are needed before completing the review....

Setting it back as draft

File: egi_notebooks_hub/d4science.py (hunk 1 above)
Comment (enolfc) [reply egi_notebooks_hub/d4science.py:44]: done

File: egi_notebooks_hub/d4science.py [hunk 2]
```diff
@@ -175,6 +190,12 @@ class D4ScienceOauthenticator(GenericOAuthenticator):
         config=True,
         help="""The OIDC URL for D4science""",
     )
+    d4science_infosys_url = Unicode(
+        D4SCIENCE_INFOSYS_URL,
+        config=True,
+        help="""The URL for the Information System of D4science""",
```
Comment (andrea-manzi) [inline egi_notebooks_hub/d4science.py:196]: this is more the URL for the GenericResources for JupyterHub

File: egi_notebooks_hub/d4science.py [hunk 3]
```diff
@@ -312,22 +377,94 @@ def get_args(self):
...
+                    override["cpu_limit"] = float(p["Cut"]["Cores"])
+                    cut_info.append(f"{p['Cut']['Cores']} Cores")
+                if "Memory" in p["Cut"]:
+                    override["mem_limit"] = "%(#text)s%(@unit)s" % p["Cut"]["Memory"]
+                    cut_info.append(f"{override['mem_limit']} RAM")
+                name += " - %s" % " / ".join(cut_info)
+            profile = {
+                "display_name": name,
+                "description": p.get("Info", {}).get("Description", ""),
+                "slug": p.get("AuthId"),
```
Comment (andrea-manzi) [inline egi_notebooks_hub/d4science.py:456]: better to add default param for the get

File: egi_notebooks_hub/d4science.py (hunk 2 above)
Comment (enolfc) [reply egi_notebooks_hub/d4science.py]: fixed

File: egi_notebooks_hub/d4science.py (hunk 3 above)
Comment (enolfc) [reply egi_notebooks_hub/d4science.py]: fixed

**Review by andrea-manzi — APPROVED** [review]
perfect!

**Pull request merged**